pysmiles/read_smiles.py

```diff
@@ -229,6 +220,64 @@ def read_smiles(smiles, explicit_hydrogen=False, zero_order_bonds=True,
     if current_ez and strict:
         raise ValueError('There is an unmatched stereochemical token.')
 
+    return mol, ez_isomer_pairs, created_ring_bonds
+
+def read_smiles(smiles, explicit_hydrogen=False, zero_order_bonds=True, 
+                reinterpret_aromatic=True, strict=True):
+    """
+    Parses a SMILES string.
+
+    Parameters
+    ----------
+    smiles : iterable
+        The SMILES string to parse. Should conform to the OpenSMILES
+        specification.
+    explicit_hydrogen : bool
+        Whether hydrogens should be explicit nodes in the output graph, or be
+        implicit in 'hcount' attributes.
+    zero_order_bonds : bool
+        Whether zero-order bonds (".") should be added as edges with an order of
+        0.
+    reinterpret_aromatic : bool
+        Whether aromaticity should be determined from the created molecule,
+        instead of taken from the SMILES string.
+    strict : bool
+        Whether to be more strict in accepting what is a valid SMILES string and
+        what is not.
+
+    Returns
+    -------
+    nx.Graph
+        A graph describing a molecule. Nodes will have an 'element', 'aromatic'
+        and a 'charge', and if `explicit_hydrogen` is False a 'hcount'.
+        Depending on the input, they will also have 'isotope' and 'class'
+        information.
+        Edges will have an 'order'.
+    """
+    bond_to_order = {'-': 1, '=': 2, '#': 3, '$': 4, ':': 1.5, '.': 0}
+    default_bond = 1
+    default_aromatic_bond = 1.5
+    mol, ez_isomer_pairs, ring_bonds = base_smiles_parser(smiles, strict=strict,
+                                                          node_attr='_atom_str', edge_attr='_bond_str')
+    for node in mol:
+        mol.nodes[node].update(parse_atom(mol.nodes[node]['_atom_str']))
+    for idx, jdx, attrs in ring_bonds:
+        if mol.nodes[idx].get('rs_isomer'):
+            mol.nodes[idx]['rs_isomer'][1].append(jdx)
+        if mol.nodes[jdx].get('rs_isomer'):
+            mol.nodes[jdx]['rs_isomer'][1].append(idx)
+    for edge in mol.edges:
+        if mol.edges[edge]['_bond_str']:
+            order = bond_to_order[mol.edges[edge]['_bond_str']]
+            if not order and not zero_order_bonds:
+                mol.remove_edge(*edge)
+                continue
+            mol.edges[edge]['order'] = order
+        elif mol.nodes[edge[0]].get('aromatic') and mol.nodes[edge[1]].get('aromatic'):
+            mol.edges[edge]['order'] = default_aromatic_bond
+        else:
+            mol.edges[edge]['order'] = default_bond
+
```

Comment:
If one were to make a bold suggestion: These could be put into functions as well such that one can mix and match a read_smiles interpreter function. Maybe even decorators? Something like an abstract `interpret_graph` method.

Reply:
I like the idea, but I need to think on it a little bit. How to organise the code and such. I suggest leaving that for a next PR